Spring-Content-JPA 2.9.0 nullpointer exception trying to setContent

**Describe the bug**
java.lang.NullPointerException: Cannot invoke "java.sql.ResultSet.close()" because "this.rs" is null

when trying to follow along the reference documentation for Spring Content JPA (trying to store any file in the H2 database)

**To Reproduce**
Steps to reproduce the behavior:
1) clone from https://github.com/tom-vandepoele-1/Spring-Content-JPA-demo
2) run DemoApplication class

Caused by: java.lang.NullPointerException: Cannot invoke "java.sql.ResultSet.close()" because "this.rs" is null
	at internal.org.springframework.content.jpa.io.BlobResourceOutputStream.close(BlobResourceOutputStream.java:104) ~[spring-content-jpa-2.8.0.jar:na]
	at java.base/java.io.FilterOutputStream.close(FilterOutputStream.java:188) ~[na:na]
	at org.apache.commons.io.IOUtils.closeQuietly(IOUtils.java:534) ~[commons-io-2.11.0.jar:2.11.0]
	at org.apache.commons.io.IOUtils.closeQuietly(IOUtils.java:466) ~[commons-io-2.11.0.jar:2.11.0]
	at org.apache.commons.io.IOUtils.closeQuietly(IOUtils.java:603) ~[commons-io-2.11.0.jar:2.11.0]
	at internal.org.springframework.content.jpa.repository.DefaultJpaStoreImpl.setContent(DefaultJpaStoreImpl.java:265) ~[spring-content-jpa-2.8.0.jar:na]

**Expected behavior**
File is stored in the H2 database without errors.

**Additional context**
Spring Content JPA 2.9.0
Spring Boot 2.7.11

--- a/spring-content-jpa/src/main/asciidoc/jpa.adoc
+++ b/spring-content-jpa/src/main/asciidoc/jpa.adoc
@@ -36,14 +36,12 @@ public static class ApplicationConfig {
 
   @Bean
   public DataSource dataSource() {
-
     EmbeddedDatabaseBuilder builder = new EmbeddedDatabaseBuilder();
     return builder.setType(EmbeddedDatabaseType.HSQL).build();
   }
 
   @Bean
   public LocalContainerEntityManagerFactoryBean entityManagerFactory() {
-
     HibernateJpaVendorAdapter vendorAdapter = new HibernateJpaVendorAdapter();
     vendorAdapter.setGenerateDdl(true);
 
@@ -56,25 +54,41 @@ public static class ApplicationConfig {
 
   @Bean
   public PlatformTransactionManager transactionManager() {
-
     JpaTransactionManager txManager = new JpaTransactionManager();
     txManager.setEntityManagerFactory(entityManagerFactory());
     return txManager;
   }
-	
+
+  @Value("/org/springframework/content/jpa/schema-drop-hsqldb.sql")
+  private Resource dropRepositoryTables;
+
+  @Value("/org/springframework/content/jpa/schema-hsqldb.sql")
+  private Resource createRepositoryTables;
+
+  @Bean
+  DataSourceInitializer datasourceInitializer() {
+    ResourceDatabasePopulator databasePopulator = new ResourceDatabasePopulator();
+
+    databasePopulator.addScript(dropRepositoryTables);
+    databasePopulator.addScript(createRepositoryTables);
+    databasePopulator.setIgnoreFailedDrops(true);
+
+    DataSourceInitializer initializer = new DataSourceInitializer();
+    initializer.setDataSource(dataSource());
+    initializer.setDatabasePopulator(databasePopulator);
+
+    return initializer;
+  }
 }
 ----
 ====
 
-This configuration class sets up an embedded HSQL database using the EmbeddedDatabaseBuilder API from spring-jdbc.  We
-then set up an EntityManagerFactory and use Hibernate as the persistence provider.  The last infrastructure component
-declared here is the JpaTransactionManager.
+This configuration class sets up an embedded HSQL database.  The first three beans are standard for any spring JPA application; `dataSource`, `entityManageFactory` and `transactionManager`.
+
+The last bean prepares the database to be able to store content by creating database schema.  Replace the `dropRepositoryTables` and `createRepositoryTables` resources with the appropriate ones for your database (`h2`, `hsqldb`, `mysql`, `oracle`, `postgresql`, `sqlserver`).  Creating this bean manually is only requires when using `spring-content-jpa` directly.  The spring-content-jpa boot starter will create this bean if not present.
 
 We activate Spring Data JPA repositories using the `@EnableJpaRepositories` annotation.  We activate Spring Content JPA
-stores using `@EnableJpaStores` and enable transaction management with the `@EnableTransactionManagement` annotation.
-
-If no base packages are configured both Spring Data JPA and Spring Content JPA will use the package that the
-configuration class resides in as the base package.
+stores using `@EnableJpaStores` (also only required when not using the boot starter) and enable transaction management with the `@EnableTransactionManagement` annotation.
 
 == Accessing Content
 

